### DISCUSSION TITLE
No module named 'solcx'

### DISCUSSION THREAD

**Discussion by Sha3-sol** [smartcontractkit/full-blockchain-solidity-course-py#112] [Q&A] [answered]
``I have followed the correction in the github along with the previous discussion, although I am still getting the same error.

Below is my code

`from solcx import compile_standard, install_solc

install_solc("0.6.0")

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)
print(compiled_sol)
`

**Comment by msperry2014** [comment]
This happened to me too - but see [#5](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/5) for an answer.

**Reply by Sha3-sol** [reply]
I am still getting this error, I really do not understand 

`import solcx

solcx.install_solc("0.8.7")

from solcx import compile_standard, install_solc

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.8.7",
)
print(compiled_sol)
`

**Comment by msperry2014** [comment]
For me, I had to create a new Python file in my project, then enter these two lines:
`import solcx`
`solcx.install_solc('0.6.0') # or whatever version we want`
Then I saved the file as 'import-solcx.py', then using the command line I entered `python import-solcx.py`

**Reply by Sha3-sol** [reply]
Thanks a lot 
But For me it still doesn't work :(

**Comment by AlbertoSinigaglia** [comment]
please post a your code formatted (since python is indentation dependent) using three back-ticks  (`), also please post the error

**Comment by PatrickAlphaC** [comment]
Did you run `pip install py-solc-x`?

**Comment by spencergoff** [comment, marked as answer]
I resolved this issue by using the command python3 deploy.py instead of python deploy.py .

**Reply by cromewar** [reply]
You are amazing.
